src/main/java/com/walmartlabs/x12/util/SegmentUtil.java

```diff
@@ -0,0 +1,51 @@
+package com.walmartlabs.x12.util;
+
+import com.walmartlabs.x12.X12Segment;
+
+import java.util.List;
+
+public class SegmentUtil {
```

Comment:
`SegmentsHolder`?

Reply:
hmmm... leaning toward SegmentIterator and implementing Java Iterator interface